system/lib/pthread/library_pthread.c

```diff
@@ -177,24 +177,18 @@ void emscripten_async_waitable_close(em_queued_call* call) {
 
 extern double emscripten_receive_on_main_thread_js(int functionIndex, int numCallArgs, double* args);
 extern int _emscripten_notify_thread_queue(pthread_t targetThreadId, pthread_t mainThreadId);
-extern int __pthread_create_js(struct pthread *thread, const pthread_attr_t *attr, void *(*start_routine) (void *), void *arg);
 
 static void _do_call(void* arg) {
   em_queued_call* q = (em_queued_call*)arg;
   // C function pointer
   assert(EM_FUNC_SIG_NUM_FUNC_ARGUMENTS(q->functionEnum) <= EM_QUEUED_CALL_MAX_ARGS);
   switch (q->functionEnum) {
-    case EM_PROXIED_PTHREAD_CREATE:
-      q->returnValue.i =
-        __pthread_create_js(q->args[0].vp, q->args[1].vp, q->args[2].vp, q->args[3].vp);
-      break;
-    case EM_PROXIED_CREATE_CONTEXT:
-      q->returnValue.i = emscripten_webgl_create_context(q->args[0].cp, q->args[1].vp);
-      break;
+#ifdef __EMSCRIPTEN_OFFSCREEN_CANVAS__
```

Comment:
Creating another libc variant purely for this purpose seems rather unfortunate.

Can we leave this in unconditionally for now, and submit PR that just removes the `EM_PROXIED_CREATE_CONTEXT`?   Then we can figure out how to somehow
incject this handler only when OFFSCREEN_CANVAS is enabled as a followup?
Perhaps we can do it via some kind of dependency injections of extension mechanism
for `_do_call`?

Reply:
Reverted for now with commit 815f592b19ee27da6a9adb0b81f35ab210ff27c5. I'm not sure what the best way is to handle this.